ietf/doc/views_doc.py

```diff
@@ -1972,7 +1972,7 @@ def rfcdiff_latest_json(request, name, rev=None):
             if doc.name != doc.canonical_name():
                 prev_rev = doc.rev
                 # not sure what to do if non-numeric values come back, so at least log it
-                log.assertion('doc.rfc_number().isdigit()')
+                log.assertion('doc.rfc_number().isdigit()') # .rfc_number() is expensive...
```

Comment:
Expensive, but at least cached. At the moment the value is used again in real code immediately after - could catch exceptions there instead of asserting ahead of time to reduce the risk that the assertions outlive the code they protect.

Reply:
Well, as you note, it's not clear what to do if the expectations of those functions are violated, and I view the assertions as a short term safety net checking that we didn't have broken data. We could probably just pull the asserts, but since they aren't really adding load, leaving them for awhile longer won't hurt us.